openedx/core/djangoapps/notifications/models.py

```diff
@@ -0,0 +1,64 @@
+"""
+Models for notifications
+"""
+from django.contrib.auth.models import User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.db import models
+from model_utils.models import TimeStampedModel
+
+
+class NotificationApplication(models.TextChoices):
+    """
+    Application choices where notifications are generated from
+    """
+    DISCUSSION = 'DISCUSSION'
+
+
+class NotificationType(models.TextChoices):
+    """
+    Notification type choices
+    """
+    NEW_CONTRIBUTION = 'NEW_CONTRIBUTION'
+
+
+class NotificationTypeContent:
+    """
+    Notification type content
+    """
+    NEW_CONTRIBUTION_NOTIFICATION_CONTENT = 'There is a new contribution. {new_contribution}'
+
+
+class Notification(TimeStampedModel):
+    """
+    Model to store notifications for users
+
+    .. no_pii:
+    """
+    user = models.ForeignKey(User, related_name="notifications", on_delete=models.CASCADE)
+    app_name = models.CharField(max_length=64, choices=NotificationApplication.choices)
+    notification_type = models.CharField(max_length=64, choices=NotificationType.choices)
+    content = models.CharField(max_length=1024)
+    content_context = models.JSONField(default={})
```

Comment:
@saadyousafarbi, @asadazam93, @AhtishamShahid, @muhammadadeeltajamul: I believe this introduced the following warning at LMS startup:

```
notifications.Notification.content_context: (fields.E010) JSONField default should be a callable instead of an instance so that it's not shared between all field instances.
	HINT: Use a callable instead, e.g., use `dict` instead of `{}`.
```